securing_apps/topics/cli-apps/cli-apps.adoc

```diff
@@ -0,0 +1,137 @@
+code the protocol and manage the tokens.  This means your CLI app will have to manage
+and be aware of token timeouts and refresh tokens.  The second disadvantage is
+that if there are other CLI apps using {project_name}, you'll have to relogin
+to each different CLI app that is secured by {project_name}.  No SSO on the command line.
+Finally, your app will have to be hard coded to specific credential types.  If you've
+only implemented password support and the {project_name} realm admin decides that an
+OTP will also be required to login, you will have to change your CLI application to
+support it.
+
+=== Kcinit CLI Utility
+
+To help developers solve some of the CLI security issues, the {project_name} team created the `kcinit` command line utility.
+This utility handles login with a Keycloak realm and stores and manages tokens locally.
+These tokens can then be passed along to your CLI applications as a command line switch or an environment variable.
+Using this tool is best described with a real example.  {project_name} has integration with the Kubernetes project.  Kubernetes
+has a backend REST API that is secured with bearer tokens and accessed with the command line utility `kubectl`.
+You can use `kcinit` to obtain and manage tokens and pass these tokens along via the `kubectl` switch `--token`.
+For example:
+
+----
+     kubectl --token=$(kcinit token)
+----
+
+`kcinit` would prompt you for login using a text-based interface.  `kcinit` would perform the OAuth protocol behind the
+scenes to obtain a token, and output the token to STDOUT.  Any access or refresh token that `kcinit` obtains
+is managed and stored locally.  This allows you to login once, and `kcinit` will handle whether you have to log
+in again.  It handles refreshing tokens automatically.  If you are using `kcinit` as command line SSO with other CLI apps,
+`kcinit` can also perform token exchanges to obtain the correct token for the exact REST backend your CLI app is invoking on.
+
+`kcinit` supports all out-of-the-box authenticators and required actions that come with {project_name}. Interaction is rendered
+in text to your console and doesn't require a browser.  Behind the scenes {project_name} uses a simple custom HTTP challenge
+protocol to obtain what needs to be displayed and inputted by users logging in.  All this text input and output use the same
+template engine that browser screens use.  This means that you can create specific themes for text logins and the text output
+is all internationalizable.
+
+If you have written some custom authenticators or required actions which do not support
+a text-based login, the {project_name} server will detect this automatically.  `kcinit` will be told that a browser is required
+to finish the login and try and launch a browser to complete the login.
+
+==== Setting Up Kcinit Within a Realm
+
+Since `kcinit` uses OAuth for login, you will have to create dedicated client for it within the {project_name}
```

Comment:
Instead of "you will have to" use the stronger phrase "you must".

...create a dedicated client...